app/signals/apps/media/views.py

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2024 Delta10 B.V.
+import mimetypes
+import os
+from datetime import timedelta
+
+from django.conf import settings
+from django.core import signing
+from django.http import HttpResponse
+from django.views.static import serve
+
+signer = signing.TimestampSigner()
+
+
+def download_file(request, path):
+    t = request.GET.get('t')
+    s = request.GET.get('s')
+
+    if not t or not s:
+        return HttpResponse('No signature provided', status=401)
+
+    try:
+        signer.unsign(f'{path}:{t}:{s}', max_age=timedelta(hours=1))
+    except signing.SignatureExpired:
+        return HttpResponse('Signature expired', status=401)
+    except signing.BadSignature:
+        return HttpResponse('Bad signature', status=401)
+
+    if settings.DEBUG:
+        response = serve(request, path, document_root=settings.MEDIA_ROOT, show_indexes=False)
```

Comment:
This is a little nitpicky, but this function is already using early returns, why not just return here as well instead of having the `else` block?

Reply:
I agree, refactored this.